src/Aspire.Hosting.Azure/AzureProvisioningResourceExtensions.cs

```diff
@@ -183,6 +184,40 @@ public static ProvisioningParameter AsProvisioningParameter(this ReferenceExpres
         return GetOrAddParameter(infrastructure, parameterName);
     }
 
+    /// <summary>
+    /// Encapsulates the logic for creating an existing or new <see cref="ProvisionableResource"/>
+    /// based on whether or not the <see cref="ExistingAzureResourceAnnotation" /> exists on the resource.
+    /// </summary>
+    /// <typeparam name="T">The type of <see cref="ProvisionableResource"/> to produce.</typeparam>
+    /// <param name="infrastructure">The <see cref="AzureResourceInfrastructure"/> that will contain the <see cref="ProvisionableResource"/>.</param>
+    /// <param name="createExisting">A callback to create the existing resource.</param>
+    /// <param name="createNew">A callback to create the new resource.</param>
+    /// <returns>The provisioned resource.</returns>
+    public static T CreateExistingOrNewProvisionableResource<T>(this AzureResourceInfrastructure infrastructure, Func<string, ProvisioningParameter, T> createExisting, Func<AzureResourceInfrastructure, T> createNew)
```

Comment:
I'm struggle a bit with this being a public API but it's not the end of the world I guess...

Reply:
What if we didn't make it an extension method? Then it wouldn't show up for normal developers, only for devs that were looking for it (which probably will only ever be us).

Reply:
I considered internal and IVT for this but that felt a little bi too much.

I do like the idea of not making it an extension method to hide it a bit better.